docs/user-guide/hypershift-agent-service-config.md

```diff
@@ -0,0 +1,141 @@
+
+# Create hypershift cluster
+hypershift create cluster agent --name $CLUSTERNAME --base-domain $BASEDOMAIN --pull-secret /tmp/pull-secret.json  --ssh-key $SSHKEY --agent-namespace $AGENT_NAMESPACE --namespace $AGENT_NAMESPACE
+
+# Wait for hypershift cluster (wait for available)
+oc get po -n $CLUSTER_NAMESPACE -w
+oc get hostedclusters -A -w
+
+# Generate kubeconfig for the generated spoke (hypershift) cluster 
+hypershift create kubeconfig --name $CLUSTERNAME --namespace $AGENT_NAMESPACE | sed s/admin/$CLUSTERNAME/g > /tmp/kubeconfig.$CLUSTERNAME
+
+# save the generated kubeconfig in a secret on the hub cluster
+oc -n $CLUSTERNAME create secret generic $CLUSTERNAME-kubeconfig --from-file=kubeconfig=/tmp/kubeconfig.$CLUSTERNAME
+```
+
+## Install CRDS
+```bash
+# install assisted-service CRDs on the spoke cluster
+oc --kubeconfig /tmp/kubeconfig.$CLUSTERNAME create namespace $CLUSTERNAME
+KUBECONFIG=/tmp/kubeconfig.$CLUSTERNAME ENABLE_KUBE_API=true make deploy-resources
+
+# install hive CRDs on the hub cluster
+oc apply -f hive.openshift.io_clusterdeployments.yaml
+oc apply -f hive.openshift.io_clusterimagesets.yaml
+oc apply -f metal3.io_baremetalhosts.yaml
+
+# install hive CRDs on the spoke cluster
+oc --kubeconfig /tmp/kubeconfig.$CLUSTERNAME apply -f hive.openshift.io_clusterdeployments.yaml
+oc --kubeconfig /tmp/kubeconfig.$CLUSTERNAME apply -f hive.openshift.io_clusterimagesets.yaml
+oc --kubeconfig /tmp/kubeconfig.$CLUSTERNAME apply -f metal3.io_baremetalhosts.yaml
+```
+
+## Infrastructure Operator
+### Install the operator
+```bash
+# install OLM (change versions as they progress)
+# download the installation script
+https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.22.0/install.sh -o install.sh
+
+# run the installation script
+chmod +x install.sh
+./install.sh v0.22.0
+
+# install assisted infrastructure operator
+oc create namespace $CLUSTERNAME
+operator-sdk run bundle -n assisted-installer $BUNDLE_IMAGE
+operator-sdk run bundle -n assisted-installer $BUNDLE_IMAGE
+```
+
+### Define HypershiftAgenConfigService
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: HypershiftAgentServiceConfig
+metadata:
+ name: agent-$CLUSTERNAME
+ namespace: $CLUSTERNAME
+spec:
+  databaseStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 10Gi
+  filesystemStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 20Gi
+  imageStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 10Gi
+  osImages:
+    - cpuArchitecture: x86_64
+      openshiftVersion: '4.11'
+      url: https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.11/4.11.0/rhcos-4.11.0-x86_64-live.x86_64.iso
+      version: 411.86.202207150124-0
+EOF
+```
+
+# Sanity test
+## Create an infraenv
+```bash
+oc create --context $CLUSTERNAME ns test
```

Comment:
'--context' is out of context in this section:)
can just replace it with '--kubeconfig /tmp/kubeconfig.$CLUSTERNAME'